controllers/trainingjob/trainingjob_controller_test.go

```diff
@@ -617,6 +617,296 @@ var _ = Describe("Reconciling a TrainingJob that exists", func() {
 				})
 			})
 		})
+
+		Context("TrainingJob is 'Failed' and has debug status 'InProgress'", func() {
+			var failureReason string
+			var debugJobStatus sagemaker.RuleEvaluationStatus
+
+			BeforeEach(func() {
+				expectedStatus = sagemaker.TrainingJobStatusFailed
+				failureReason = "Failure within the training job"
+				debugJobStatus = sagemaker.RuleEvaluationStatusInProgress
+
+				// Add the failure reason to the describe output
+				describeOutput := CreateDescribeOutputWithDebugJobStatus(expectedStatus, expectedSecondaryStatus, debugJobStatus)
+				describeOutput.FailureReason = ToStringPtr(failureReason)
+
+				mockSageMakerClientBuilder.
+					AddDescribeTrainingJobResponse(describeOutput)
+			})
+
+			When("!HasDeletionTimestamp", func() {
+				It("Requeues after interval", func() {
+					ExpectRequeueAfterInterval(reconcileResult, reconcileError, pollDuration)
+				})
+
+				It("Updates status", func() {
+					ExpectJobsStatusToBe(trainingJob, string(expectedStatus), string(expectedSecondaryStatus), string(debugJobStatus))
+				})
+
+				It("Has the additional field set", func() {
+					ExpectAdditionalToContain(trainingJob, failureReason)
+				})
+
+				Context("Does not have a finalizer", func() {
+					BeforeEach(func() {
+						shouldHaveFinalizer = false
+					})
+
+					It("Adds a finalizer", func() {
+						ExpectToHaveFinalizer(trainingJob, controllers.SageMakerResourceFinalizerName)
+					})
+				})
+			})
+
+			When("HasDeletionTimestamp", func() {
+				BeforeEach(func() {
+					shouldHaveDeletionTimestamp = true
+				})
+
+				It("Deletes the training job", func() {
+					ExpectTrainingJobToBeDeleted(trainingJob)
+				})
+
+				It("Requeues after interval", func() {
+					ExpectRequeueAfterInterval(reconcileResult, reconcileError, pollDuration)
+				})
+			})
+		})
+
+		Context("TrainingJob is 'Failed' and has debug job with status 'Stopping'", func() {
```

Comment:
For consistency with the other contexts can we make this `TrainingJob is 'Failed' and has debug status 'Stopping'`